pkg/kv/kvserver/store_raft.go

```diff
@@ -519,6 +533,41 @@ func (s *Store) processReady(rangeID roachpb.RangeID) {
 	}
 }
 
+type cordonRecord struct {
+	RangeID roachpb.RangeID
+}
+
+func (s *Store) markReplicaForCordoningIfPanic(ctx context.Context, rangeID roachpb.RangeID) {
+	if re := recover(); re != nil {
+		func() {
+			log.Errorf(ctx, "cordoning range %v since panic and/or fatal error during application: %v",
+				rangeID, re)
+
+			// TODO(josh): For the POC, we only allow cordoning a single range at a time.
+			c := cordonRecord{RangeID: rangeID}
+			// TODO(josh): Use proto instead of yaml.
+			mc, err := yaml.Marshal(c)
+			if err != nil {
+				log.Errorf(ctx, "couldn't marshall cordon record: %v", err)
+				return
+			}
+			// TODO(josh): I don't think we need an fsync here? The process will restart,
```

Comment:
I agree, an fsync seems unnecessary here.

Reply:
💯 

Is there an easy way to tell storage to NOT `fsync`? I can also look around myself.